### PR TITLE
Burstcoin Extension.

### PR DESCRIPTION
**Pull request by nixops**
Burst coin extension.

Burst: https://burst-coin.org

### CODE REVIEW

**Comment by harry1453** [comment]
Build should now be fixed.

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/burstcoin/BurstExtension.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                    try {
+                        rate = new BigDecimal(st.nextToken());
+                    } catch (Throwable ignored) {}
+                }
+                String preferedFiatCurrency = Currencies.USD;
+                if (st.hasMoreTokens()) {
+                    preferedFiatCurrency = st.nextToken().toUpperCase();
+                }
+                return new FixPriceRateSource(rate,preferedFiatCurrency);
+            } else if ("poloniexrs".equalsIgnoreCase(rsType)) {
```
Comment (rpanak-generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/burstcoin/BurstExtension.java]: Change prefixes as suggested in XML comments

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/burstcoin/wallets/burstwallet/BurstWallet.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    private final String publicKey;
+    private final String accountId;
+
+    private final BurstWalletAPI api;
+
+    public BurstWallet(String masterPassword, String accountId) {
+        this.masterPassword = masterPassword;
+        this.publicKey = BurstCryptoUtils.getPublicKey(masterPassword);
+        this.accountId = accountId;
+        this.api = RestProxyFactory.createProxy(BurstWalletAPI.class, "https://wallet.burst.cryptoguru.org:8125");
```
Comment (rpanak-generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/burstcoin/wallets/burstwallet/BurstWallet.java]: Make URL configurable

**Comment by harry1453** [comment]
All done :)

**Comment by rico666** [comment]
Tak co? Dobrý - nebo stále něco chybí?

**Comment by rpanak-generalbytes** [comment]
What's the source of Curve25519.java? I've tried to find it but couldn't find the exact version you submitted. Depending on the version you use, there might be some problem in to code - see [NXT review](https://gist.github.com/doctorevil/9521116).

**Comment by rico666** [comment]
The Curve25519.java sign-fix was patched March, 15th 2014 into NXT.

See https://github.com/mrv777/NXT/commit/56e44b38e524ea52a7b2f5a7e25d126c929ece72#diff-3f210d560753a3100a17d79ef460a9a1

Burst forked off from NXT in August 2014 using already the patched version.

See today: https://github.com/PoC-Consortium/burstcoin/blob/master/src/brs/crypto/Curve25519.java#L106

So as far as we're concerned, there are no issues with that code.

**Comment by harry1453** [comment]
I have performed fairly thorough testing of functions that utilize the Curve25519 class on both testnet and mainnet using multiple passphrase/account combinations, including checking the signatures produced by the `sign()` helper method against those produced by the Burst Reference Software (BRS), the source of which Rico linked to above. The class is not an exact match as I removed functions that were unused in this implementation in the interest of simplifying the codebase. These changes have been tested.

**Comment by rpanak-generalbytes** [comment]
Hi all,

I've explicitly mentioned the origin of the file in the comment. I think we're good to go...

Thanks for the PR. Merging.

**Pull request merged**